### ISSUE TITLE
Rewrite CLI options handling

### ISSUE THREAD

**Issue by kittens** [babel/babel#859]
Will need to either roll our own or switch to an already existing solution.

**Comment by Raynes** [comment]
I've noticed that the initial step of babel-node that plucks out node arguments picks them up from anywhere. This isn't a commander issue, since this is before the args even get to commander.

**Comment by sebmck** [comment]
@Raynes Yep. It's hard to juggle two different sets of arguments since the babel-specific ones need to be consumed while all others need to be passed on.

**Comment by sebmck** [comment]
Superseded by #883.

**Comment by Rovack** [comment]
@thejameskyle, @kittens: Hello. I recently encountered the same problem a few people have reported with passing flags through babel-node, and was wondering what the background was behind closing this ticket?
As far as I can tell, https://github.com/babel/babel/issues/836 
and https://github.com/babel/babel/issues/1730, which were supposed to be resolved as part of this, still pose a problem today, even though they were first opened 2 years ago.

It appears this was all originally supposed to be resolved as part of https://github.com/babel/babel/issues/883 or https://github.com/babel/babel/issues/988, but those issues were eventually dropped and this was reopened.
Nonetheless, it looks like no changes were actually made before this was closed again.

So, are there currently no plans to fix babel-node being, as it stands, positionally insensitive?
